src/Controllers/ShareDraftController.php

```diff
@@ -119,18 +120,26 @@ public function preview(HTTPRequest $request)
      *
      * @return HTTPResponse
      */
-    protected function getRenderedPageByURLSegment($url)
+    protected function getRenderedPageByURL($url)
     {
-        $pageRequest = HTTPRequestBuilder::createFromEnvironment();
-        $pageRequest->setHttpMethod('GET');
-        $pageRequest->setUrl($url);
+        // Clean and update live global variables
+        $variables = HTTPRequestBuilder::cleanEnvironment(Environment::getVariables());
```

Comment:
Put in a comment that you're basically doing what happens in `HTTPRequestBuilder::createFromEnvironment()`, I was pretty confused by the `@file_get_contents('php://input')` at first

Reply:
@emteknetnz Done.